### PR TITLE
RavenDB-6012 Implement facets for the new client

### PR DESCRIPTION
**Pull request by efratshenhar**

### CODE REVIEW

File: src/Raven.NewClient/Commands/GetFacetsCommand.cs [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+                throw new InvalidOperationException($"You cannot specify both '{nameof(FacetQuery.FacetSetupDoc)}' and '{nameof(FacetQuery.Facets)}'.");
+
+            var method = Query.CalculateHttpMethod();
+
+            var request = new HttpRequestMessage
+            {
+                Method = method
+            };
+            
+            if (method == HttpMethod.Post)
```
Comment (ayende) [inline src/Raven.NewClient/Commands/GetFacetsCommand.cs:34]: Isn't it always POST now?

File: src/Raven.NewClient/Document/DocumentConvention.cs [hunk 2]
```diff
@@ -542,6 +543,7 @@ public JsonSerializer CreateSerializer()
             };
 
             CustomizeJsonSerializer(jsonSerializer);
+            jsonSerializer.Converters.Add(new StringEnumConverter());
```
Comment (ayende) [inline src/Raven.NewClient/Document/DocumentConvention.cs:546]: Why do we need this?  This is json.net, and it is going away, no?

Comment (efratshenhar) [reply src/Raven.NewClient/Document/DocumentConvention.cs:546]: We need this for enum to be string and not int . i will change it not to use  json.net.

File: src/Raven.NewClient/Commands/GetFacetsCommand.cs (hunk 1 above)
Comment (efratshenhar) [reply src/Raven.NewClient/Commands/GetFacetsCommand.cs:34]: yes, will be fixed

File: src/Raven.NewClient/Document/DocumentConvention.cs (hunk 2 above)
Comment (ayende) [reply src/Raven.NewClient/Document/DocumentConvention.cs:546]: Yes, but that should be handled by the blittable, not json.net

**Pull request merged**